packages/xo-web/src/common/sorted-table/index.js

```diff
@@ -209,29 +209,23 @@ class IndividualAction extends Component {
       isFunction(disabled) ? disabled(item, userData) : disabled
   )
 
-  _redirect = createSelector(
-    () => this.props.item,
-    () => this.props.redirect,
-    () => this.props.userData,
-    (item, redirect, userData) =>
-      isFunction(redirect) ? redirect(item, userData) : redirect
-  )
-
   _executeAction = () => {
     const p = this.props
-    return p.handler && p.handler(p.item, p.userData)
+    return p.redirectOnSuccess
+      ? { item: p.item, userData: p.userData }
```

Comment:
But if there's a `handler`, we still want to run it first. And you can also add a comment since it's not obvious why you're returning this in `_executeAction`.